### PR TITLE
[WIP] Create a platform/cli dir and add a micro signup command

### PR DESCRIPTION
**Pull request by asim**
- Create a platform/cli package for anything plaform specific
- Move the Signup function to platform/cli so `micro login` function only does platform.Signup(ctx) when --otp is set
- Create a `micro signup` command that does exactly what `micro login --otp` does verbatim.

### CODE REVIEW

**Comment by crufter** [comment]
Looking good so far. Gotta modify the `m3o_test.go` too as it uses `micro login --otp`.

**Comment by asim** [comment]
We'll also need to ask for password during sign-up and pass to sign-up completion if we want to do it the right way. At that point I think login --otp would probably go away or go via login service...

File: service/auth/auth.go [hunk 1]
```diff
@@ -314,8 +318,8 @@ func Commands(srvOpts ...micro.Option) []*cli.Command {
 					Usage: "Login/signup with a One Time Password.",
 				},
 				&cli.StringFlag{
-					Name:  "email",
-					Usage: "Email address to use for Login with OTP",
+					Name:  "password",
+					Usage: "Password for automated logins",
```
Comment (asim) [inline service/auth/auth.go:322]: "Plain text password"

**Comment by asim** [comment]
Closing in favour of https://github.com/micro/micro/pull/1071

**Pull request closed without merging**